src/Files.App/Commands/IRichCommand.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.UI.Xaml.Input;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using System.Windows.Input;
+
+namespace Files.App.Commands
+{
+	public interface IRichCommand : ICommand, INotifyPropertyChanging, INotifyPropertyChanged
+	{
+		CommandCodes Code { get; }
+
+		string Label { get; }
```

Comment:
Should we have a separate string for automation properties? Maybe one for tooltips as well.

Reply:
I notice that the automation.name always have the same text as the label. Concerning the particular case of commands, there is no reason for automation.name to be different from the name of the command. You can specify that it is a command (Command Show file extensions). This makes it possible not to complicate the actions, it suffices to add a property to RichCommand.

For the tooltip, it will probably be necessary to add a Description property to the actions. It will probably also be useful for UIs to manage commands (hotkeys, and maybe future custom menus).

Does this fit?

Reply:
That's in line with what I'm thinking. Perhaps we can automatically generate the tooltip by combining the label and hotkey.

Reply:
It's done like this in the draft, there is a LabelWithHotKey property at IRichCommand. This is not the case here because the hotkeys will come later.